### PR TITLE
Update package

### PR DESCRIPTION
**Pull request by ubaldot**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

This is based on this suggestion: https://github.com/conda-forge/qtserialport-feedstock/commit/fc7a3538deba400a92f7b5572c8dae2712d5584a

However, I haven't understand if I shall also create a new file `.ci_support/migrations/qt515.yaml` in this repo.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by ubaldot** [comment]
@hmaarrfk I just want to mention this https://github.com/conda-forge/qtserialport-feedstock/issues/7 and therefore this PR may be closed?

**Comment by hmaarrfk** [comment]
my inspection is showing that this is included in the qt5-main package but isn't included in the qt6-main package.

We are trying to leave the qt6-main package a little slimmer.

So it may be appropriate to update this package for qt6 only.

**Comment by ubaldot** [comment]
Ah ok! It was a bit difficult to find! :D I used `conda search qt` and I observed that qt 6.1.0  was available for anaconda but qt 5.15.8 was available for conda-forge. But I didn't know I could `conda install qt6-main`. 

Yes, it may be appropriate to update only for this package, but then I think it may be confusing: we have the "standard" qt6 that you can install with e.g. `homebrew qt6` and it may include the serial, `conda install -c anaconda qt` would install the qt6 perhaps with the serial module included and one should `conda install -c conda-forge qt6-main` and `conda install -c conda-forge qtserial port` to get the same result - if I understood it correctly.

**Comment by hmaarrfk** [comment]
A few things:

1. Re homebrew: Different packaging repositories are free to make tradeoffs between how things are put together. I'm not going to discuss the tradeoffs in a PR like this. Qt is notoriously difficult to package. Please see fuller discussion in the various issues: https://github.com/conda-forge/qt-main-feedstock/issues
2. Re usage of `-c anaconda`: we just don't support this at conda forge. We are a volunteer run community and are not directly responsible for things on the anaconda channel. Our packages have not been compatible with them for a while. Please find a way to contact Anaconda if you have questions on usage on their packages.


We have diverged on quite a few packages from the above two package management solutions. If you feel like conda-forge is useful for you, please consider contributing, if you feel like you are bettered suited by a combination of homebrew + anaconda channel, feel free to close these PRs.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,38 +1,42 @@
...
   # QtSerialPort already comes with Windows Qt builds but the included bld.bat
   # should work (just doesn't package anything because files already exist)
   skip: true  # [win]
-  number: 1
+  number: 0
 
 requirements:
   build:
     - perl
     - cmake
```
Comment (hmaarrfk) [inline recipe/meta.yaml:22]:
```suggestion
    - make  # [unix]
    - cmake
```

**Comment by hmaarrfk** [comment]
Ultimately, to make your software "most compatible" for end users you have a few paths:

1. You create your own channel that is compatible with a strong base (like anaconda's channel). This is the strategy that pytorch and tensorflow's official packages take.
   * your users have to enable anaconda's channel's + yours to get your package.
2. You can work to contribute packages to a community run project so that it its compatible with the community's defined ecosystem:
   * Your users enable conda-forge's channel. Then just install your package.

You can therefore just try to make a package for your software, and see if all the requirements are satisfied by conda-forge' channel.

We share alot of ideas with the anaconda channel, just we have different priorities. so sometimes ideas materialize there first, and vice versa.

**Comment by ubaldot** [comment]
Ah ok! Thanks for the clarification. I am fine with conda-forge actually. :) 
Obviously if I have an issue with X I would directly contact X. 
I was just thinking out loud, of course you can diverge as much as you want from other package management systems :)

**Comment by hmaarrfk** [comment]
just trying to keep the conversation focused and constructive on the next steps.

Your next step may be to submit a recipe to staged recipes. feel free to ping me if you've tried for a few hours to get it to work and still can't.

**Comment by ubaldot** [comment]
I actually already placed a recipe in staged-recipes: https://github.com/conda-forge/dymoval-feedstock :) I also added [this](https://github.com/conda-forge/sfml-feedstock/pull/7) as a step to the ultimate goal to learn packing non-python code with conda :)

**Comment by hmaarrfk** [comment]
ah yeah. ok do ping me if you have questions.

You can try to update this to be qt6, i added a test that should "fail" in qt5 but "pass in qt6" during the building step to justify the continued maintenance of this package.

**Comment by ubaldot** [comment]
> You can try to update this to be qt6 

I am not sure I understand what you mean. How should I try to update this feedstock to qt6?

File: recipe/build.sh [hunk 2]
```diff
@@ -1,8 +1,12 @@
 #!/bin/bash
+set -ex
 
 [[ -d build ]] || mkdir build
 cd build/
 
+test ! -f ${PREFIX}/lib/libQt5SerialPort${SHLIB_EXT} || (
+   echo libQt5SerialPort already exists this package does not need to be built && exit $?
```
Comment (hmaarrfk) [inline recipe/build.sh]:
```suggestion
   echo libQt5SerialPort already exists this package does not need to be built && exit 1
```

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
Honestly, i'm just copying jschueller's work from 
https://github.com/conda-forge/qt-charts-feedstock/blob/main/recipe/meta.yaml

**Pull request closed without merging**